File: app/pong/game.py
import pygame
import random
# from app.db.sql import *

# Initialize pygame
pygame.init()

# Set up the game window
WIDTH = 610
HEIGHT = 600
win = pygame.display.set_mode((WIDTH, HEIGHT))
pygame.display.set_caption("Pong Game")

background_image = pygame.image.load("data/background.png")
background_rect = background_image.get_rect()

# Define colors
WHITE = (255, 255, 255)
YELLOW = (254, 245, 164)
RED = (236, 123, 125)

# Set up the game clock
clock = pygame.time.Clock()

# Set up the paddle
paddle_width = 100
paddle_height = 20
paddle_x = (WIDTH - paddle_width) // 2
paddle_y = HEIGHT - paddle_height
paddle_speed = 5

ball_image = pygame.transform.scale(pygame.image.load("data/kirbyball.png"),(40,40))  # Replace with the path to your ball image
ball_radius = 10
# Set up the ball
ball_rect = ball_image.get_rect()
ball_rect.center = (WIDTH // 2, HEIGHT // 2)
ball_dx = random.choice([-2, 2])
ball_dy = -2

# Set up blocks
block_width = 50
block_height = 20
block_rows = 5
block_cols = 10
block_margin = 10
blocks = []
for row in range(block_rows):
    for col in range(block_cols):
        block_x = col * (block_width + block_margin) + block_margin
        block_y = row * (block_height + block_margin) + block_margin
        blocks.append(pygame.Rect(block_x, block_y, block_width, block_height))

score = 0

# Game loop
running = True
while running:
    # Handle events
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    win.blit(background_image, background_rect)

    # Move the paddle
    keys = pygame.key.get_pressed()
    if keys[pygame.K_LEFT] and paddle_x > 0:
        paddle_x -= paddle_speed
    if keys[pygame.K_RIGHT] and paddle_x < WIDTH - paddle_width:
        paddle_x += paddle_speed

    # Move the ball
    ball_rect.x += ball_dx
    ball_rect.y += ball_dy

    # Check ball collision with paddle
    if ball_rect.y + ball_radius > paddle_y and ball_rect.x + ball_radius > paddle_x and ball_rect.x - ball_radius < paddle_x + paddle_width:
        ball_dy = -ball_dy

    # Check ball collision with blocks
    for block in blocks:
        if block.collidepoint(ball_rect.x, ball_rect.y):
            blocks.remove(block)
            score += 1
            ball_dy = -ball_dy
            break

    # Check ball collision with walls
    if ball_rect.x - ball_radius < 0 or ball_rect.x + ball_radius > WIDTH:
        ball_dx = -ball_dx
    if ball_rect.y - ball_radius < 0:
        ball_dy = -ball_dy
    elif ball_rect.y + ball_radius > HEIGHT:
        running = False  # Game over if ball hits the bottom

    # Clear the screen
    #win.fill(WHITE)

    # Draw the paddle
    pygame.draw.rect(win, RED, (paddle_x, paddle_y, paddle_width, paddle_height))

    # Draw the ball
    win.blit(ball_image, ball_rect)

    # Draw the blocks
    for block in blocks:
        pygame.draw.rect(win, YELLOW, block)

    # Display the score
    font = pygame.font.Font(None,36)
    text = font.render("Score: " + str(score), True, YELLOW)
    win.blit(text, (WIDTH-110, HEIGHT - 40))

    # Update the display
    pygame.display.update()

    # Limit the frame rate
    clock.tick(60)

# Quit the game

def add_score(user, game, score):
  c = db.cursor()
  c.execute("select game from users where (user=?)")
  newScore = float(c.fetchone())
  newScore += score
  c.execute("REPLACE INTO table(game) where username=user VALUES(score)")
  db.commit()
  c.close()

session_user = F"{get_username(session['ID'])}"
add_score(session_user, "pong_high", score)
pygame.quit()

File: app/pong/game.js
// Set up the canvas
const canvas = document.getElementById("gameCanvas");
const ctx = canvas.getContext("2d");

// Set up the paddle
const paddleWidth = 100;
const paddleHeight = 10;
let paddleX = (canvas.width - paddleWidth) / 2;
const paddleY = canvas.height - paddleHeight - 10;
const paddleSpeed = 5;

// Set up the ball
const ballRadius = 10;
const ballImage = newImage()
ballImage.src = "data/kirbyball.png";
let ballX = Math.random() * (canvas.width - ballRadius * 2) + ballRadius;
let ballY = canvas.height / 2;
let ballDX = Math.random() > 0.5 ? -2 : 2;
let ballDY = -2;

// Set up blocks
const blockWidth = 50;
const blockHeight = 20;
const blockRows = 5;
const blockCols = 10;
const blockMargin = 10;
let blocks = [];
for (let row = 0; row < blockRows; row++) {
  for (let col = 0; col < blockCols; col++) {
    const blockX = col * (blockWidth + blockMargin) + blockMargin;
    const blockY = row * (blockHeight + blockMargin) + blockMargin;
    blocks.push({ x: blockX, y: blockY, width: blockWidth, height: blockHeight });
  }
}

const backgroundImage = new Image();
backgroundImage.src = "data/background.png"

let score = 0;

// Move the paddle
function movePaddle(e) {
  if (e.key === "ArrowLeft" && paddleX > 0) {
    paddleX -= paddleSpeed;
  } else if (e.key === "ArrowRight" && paddleX < canvas.width - paddleWidth) {
    paddleX += paddleSpeed;
  }
}

document.addEventListener("keydown", movePaddle);

// Game loop
function draw() {
  // Clear the canvas
  ctx.clearRect(0, 0, canvas.width, canvas.height);

  // Move the ball
  ballX += ballDX;
  ballY += ballDY;

  // Draw the paddle
  ctx.fillStyle = "#0000FF";
  ctx.fillRect(paddleX, paddleY, paddleWidth, paddleHeight);

  //background
  ctx.drawImage(backgroundImage, 0, 0, canvas.width, canvas.height);

  // Draw the ball
  ctx.drawImage(pygame.transform.scale(ballImage,(40,40)), ballX - ballWidth / 2, ballY - ballHeight / 2, ballWidth, ballHeight);
  ctx.beginPath();
  ctx.arc(ballX, ballY, ballRadius, 0, Math.PI * 2);
  ctx.fillStyle = "#FF0000";
  ctx.fill();
  ctx.closePath();

  // Draw the blocks
  ctx.fillStyle = "#0000FF";
  blocks.forEach((block) => {
    ctx.fillRect(block.x, block.y, block.width, block.height);
  });

  // Display the score
  ctx.fillStyle = "#0000FF";
  ctx.font = "24px Arial";
  ctx.fillText("Score: " + score, 10, 30);

  // Check ball collision with paddle
  if (
    ballY + ballRadius > paddleY &&
    ballX + ballRadius > paddleX &&
    ballX - ballRadius < paddleX + paddleWidth
  ) {
    ballDY = -ballDY;
  }

  // Check ball collision with blocks
  blocks.forEach((block, index) => {
    if (
      ballY - ballRadius < block.y + block.height &&
      ballX > block.x &&
      ballX < block.x + block.width
    ) {
      blocks.splice(index, 1);
      score++;
      ballDY = -ballDY;
    }
  });

  // Check ball collision with walls
  if (ballX - ballRadius < 0 || ballX + ballRadius > canvas.width) {
    ballDX = -ballDX;
  }
  if (ballY - ballRadius < 0) {
    ballDY = -ballDY;
  } else if (ballY + ballRadius > canvas.height) {
    // Game over if ball hits the bottom
    document.removeEventListener("keydown", movePaddle);
    return;
  }

  requestAnimationFrame(draw);
}

draw();
